### PR TITLE
chore: Extract entity num generation from HandleContext

### LINKED ISSUE #13932: Extract EntityNum creator

### PR DESCRIPTION
**Pull request by netopyr**
**Description**:

This PR extracts the entity num generation from HandleContext

**Related issue(s)**:

Fixes #13932 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/ids/EntityNumGeneratorImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.spi.ids.EntityNumGenerator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+
+/**
+ * Default implementation of {@link EntityNumGenerator}.
+ */
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/ids/EntityNumGeneratorImpl.java:27]: We can annotate this with @ChildDispatchScope

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM! Thanks @netopyr

**Review by povolev15 — APPROVED** [review]
LGTM TY @netopyr

**Review by jeromy-cannon — APPROVED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Pull request merged**